tags.go

```diff
@@ -26,10 +26,18 @@ func DefaultTagSuffix(ref, suffix string) []string {
 
 // DefaultTags returns a set of default suggested tags based on
 // the commit ref.
-func DefaultTags(ref string) []string {
+func DefaultTags(ref, defaultBranch string) []string {
+
+	if defaultBranch != "" && strings.HasPrefix(ref, "refs/heads/") {
+		if stripHeadPrefix(ref) != defaultBranch {
```

Comment:
I think we should just use `if strings.TrimPrefix(ref, "refs/heads") != defaultBranch`. I'm not sure we need a helper function since this is not used anywhere else and it is a 1-liner.

Reply:
We need to make sure the ref is branch or tag so it is not working for using `strings.TrimPrefix(ref, "refs/heads") != defaultBranch`

Reply:
`if defaultBranch != ""` should not be required because `defaultBranch` will never be empty

Reply:
but really I think this whole block should be removed, since we should never return an empty tag. If there is a goal (not publishing images for branches) we can discuss this, but returning an empty tag is not the way I would like to enforce that logic.